### PR TITLE
Reviewer Matt - Add debian packaging infrastructure to the v1.6 branch

### PR DESCRIPTION
**Pull request by bossmc**
The contents of the folder are based on the debian folder from the current v1.4 memcached in the ubuntu repos (apt-get source memcached).

I've tested by building a package and installing on a clean ubuntu 12.04 system.

The change to default_engine.c allows the memcached to be tapped more than 10 times in series (it can't be tapped more than 10 times in parallel for other reasons).

### CODE REVIEW

**Comment by mirw** [comment]
@bossmc, I think you asked me to hold off on reviewing this while you fixed an issue with it.  Please an you let me know if I've misunderstood or, if not, when I'm good to go again?

**Comment by bossmc** [comment]
Ok, memcached start-up is fixed, please review.

File: debian/patches/01_init_script_additions.patch [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
++               echo "$NAME."
++       else
++            echo "$NAME disabled in /etc/default/memcached."
++       fi
++       ;;
++  status)
++       $START_PREFIX status_of_proc -p $PIDFILE $DAEMON $NAME  && exit 0 || exit $?
++       ;;
++  *)
++    N=/etc/init.d/$NAME
```
Comment (mirw) [inline debian/patches/01_init_script_additions.patch]: (Pedantic) Indentation.

**Comment by mirw** [comment]
Looks good.  One minor comment above.

**Pull request merged**